### ISSUE TITLE
Tauri/Tauri Specta v2 known issues

### ISSUE THREAD

**Issue by oscartbeaumont** [oscartbeaumont/tauri-specta#86]
### To reproduce
```bash
pnpm i
cd examples/apps
pnpm tauri dev
```

#### Crash

Press "Send Event" button and the app crashes:
```
thread 'main' has overflowed its stack
fatal runtime error: stack overflow
```

### To reproduce
```bash
pnpm i
cd examples/custom-plugin/app
pnpm tauri dev
```

#### Permissions for plugins

I'm using `tauri_plugin` within a `build.rs` however I am getting the following error:
```
Caused by:
  process didn't exit successfully: `/Users/oscar/Desktop/tauri-specta/target/debug/build/tauri-specta-example-custom-plugin-51c83bbcfcceee66/build-script-build` (exit status: 1)
  --- stdout
  tauri-plugin: expected build script env var CARGO_MANIFEST_LINKS, but it was not found - ensure this is called in a build script
warning: build failed, waiting for other jobs to finish...
```

#### Other observations

 - Tauri migrate tool failed.
    - Run it in `examples/app` on this commit https://github.com/oscartbeaumont/tauri-specta/commit/4e48b7b378ce89590a9e9befe29a83b0cd4b322a to reproduce.
 - Sometimes clicks into the Tauri window just don't work and my macOS menubar showed "Code" or "Spotify" not the Tauri app.

**Comment by amrbashir** [comment]
> I'm using [`tauri-plugin`](https://docs.rs/tauri-plugin) within `examples/custom-plugin/plugin/build.rs`, however I am getting the following error:

The plugin is missing this line in Cargo.toml
```rs
links = "tauri-specta-example-custom-plugin"
```
see https://github.com/tauri-apps/plugins-workspace/blob/f39586bcb44ba13aca16d1b7ade77ce3d1d84982/plugins/fs/Cargo.toml#L10

also iirc, I think we require that the plugin crate name be prefixed with `tauri-plugin`.

> 


> #### `ManagerExt` trait
> We had to introduce a [`ManagerExt`](https://github.com/oscartbeaumont/tauri-specta/blob/c34c1ef66c5575368425868008e85d0850bcc762/src/manager_ext.rs#L7) trait as `tauri::Manager` does not have `listen`, `once` and `unlisten`.
> 
> It would be great to either merge this into `tauri::Manager` or upstream it as `tauri::ManagerExt` as this may be generally useful to other ecosystem packages.

I guess that's fine to upstream, we lose the customized documentation but that should be okay.

**Comment by amrbashir** [comment]
> * Sometimes clicking into the Tauri window just won't work until I refocus another app and try again. It shows "Code" or "Spotify" or "Discord" on the macOS menubar instead of the Tauri app. This is hard to reproduce because it's been rare but I could it in video. In the following video I am continually clicking my mouse and the Tauri window is not taking focus and the webview buttons are not reacting.

Can't help with this one so I will ping @pewsheen if he has any idea

**Comment by oscartbeaumont** [comment]
> The plugin is missing this line in Cargo.toml

Okay I added the `links` property and managed to fix a couple of other compile-time errors but now i'm getting this when I execute the command defined by the plugin:
<img width="485" alt="Screenshot 2024-04-30 at 10 33 31 PM" src="https://github.com/oscartbeaumont/tauri-specta/assets/21004798/5f72496f-6a2f-4728-842f-ce4affcb3d60">

Screenshot taken in the webview devtools.

> `ManagerExt` trait

If I did it as a PR, do you recon the methods would be better directly on the `Manager` trait or just introducing the new `ManagerExt` trait? Both options should be technically viable.

I personally would lean towards the `Manager` as it matches the Tauri v1 behavior closer but I wonder if removing these methods from it was actually intentional or if it was just lost unintentionally in other changes.

It's completely fine if this isn't upstreamed too, I just think it's probably gonna be useful outside of Tauri Specta, athough its utility is probably going to be niche so that would be a fair argument against it.

> Any particular errors logged?

```bash
git clone https://github.com/oscartbeaumont/tauri-specta
cd tauri-specta
git checkout 4e48b7b378ce89590a9e9befe29a83b0cd4b322a
pnpm i
cd examples/app/
npx @tauri-apps/cli@next migrate
```

```
➜ npx @tauri-apps/cli@next migrate
    Info Replacing `@tauri-apps/api/window` with `@tauri-apps/api/webviewWindow` on /Users/oscar/Desktop/ts2/examples/app/src/main.ts
    Info Replacing `@tauri-apps/api/tauri` with `@tauri-apps/api/core` on /Users/oscar/Desktop/ts2/examples/app/src/bindings.ts
    Info Replacing `@tauri-apps/api/window` with `@tauri-apps/api/webviewWindow` on /Users/oscar/Desktop/ts2/examples/app/src/bindings.ts
    Info Installing Cargo dependency "tauri-plugin-http"...
    Error Could not migrate plugin 'http': Failed to install Cargo dependency
```

I feel like I recall it failing on the CLI plugin last time but don't quote me on that.

p.s. thanks @amrbashir for taking the time to help!

**Comment by amrbashir** [comment]
> > The plugin is missing this line in Cargo.toml
> 
> Okay I added the `links` property and managed to fix a couple of other compile-time errors but now i'm getting this when I execute the command defined by the plugin: <img alt="Screenshot 2024-04-30 at 10 33 31 PM" width="485" src="https://private-user-images.githubusercontent.com/21004798/326850028-5f72496f-6a2f-4728-842f-ce4affcb3d60.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ0ODkyNDIsIm5iZiI6MTcxNDQ4ODk0MiwicGF0aCI6Ii8yMTAwNDc5OC8zMjY4NTAwMjgtNWY3MjQ5NmYtNmEyZi00NzI4LTg0MmYtY2U0YWZmY2IzZDYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDMwVDE0NTU0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4NmRkY2ViZjkzM2M4ZGFmZjEyYmY2ZWU3OGNmMWU0YTI3MGFiYmIzZTc0MWY0OGFiZjljZmM4ZDQ2MjI4NzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.EUbQHzfMtmPLjDE71ggu3FVG0HPL_FTdcNgb_Ykqr1M">
> 
> Screenshot taken in the webview devtools.
> 

I think that's because the plugin is named in `tauri-plugin-specta-example` in Cargo.toml but then registered as `custom-plugin` in `lib.rs`, maybe use one the same name.

**Comment by oscartbeaumont** [comment]
> the only relevant changes to events in the Manager trait between v1 and v2

Yeah, your completely right, my bad for getting lost in all the changes.

> About the migration errors,

Ahhh, okay makes sense.

> I think that's because the plugin is named in

Yeah, that was it thanks for your help!

**Comment by oscartbeaumont** [comment]
I have opened the PR to Tauri about the manager stuff - https://github.com/tauri-apps/tauri/pull/9627.

@amrbashir do you have an idea of when the next Tauri release is going to happen?
Tauri Specta right now requires a Git patch for [#9392](https://github.com/tauri-apps/tauri/pull/9392) which prevents me from publishing to crates.io.

Thanks again for taking the time to help!

**Comment by amrbashir** [comment]
That's already published https://github.com/tauri-apps/tauri/releases/tag/tauri-v2.0.0-beta.17

**Comment by oscartbeaumont** [comment]
Sick!

**Comment by oscartbeaumont** [comment]
I have just published Tauri Specta v2.0.0-rc.5 but the build has failed on docs.rs with some issues about javascriptcore and soup3.

Do you know if Tauri did anything special to workaround this?

[build logs](https://docs.rs/crate/tauri-specta/2.0.0-rc.5/builds/1206568)

**Comment by amrbashir** [comment]
yes you need to activate `docsrs` flag when building documentation see https://github.com/tauri-apps/tauri/blob/cd5580749c285207679678106642381c18e6597b/core/tauri/Cargo.toml#L29-L30
